source/extensions/filters/http/gcp_authn/gcp_authn_filter.cc

```diff
@@ -74,8 +74,7 @@ Http::FilterHeadersStatus GcpAuthnFilter::decodeHeaders(Http::RequestHeaderMap&
     // "http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/identity?audience=[AUDIENCE]"
     // So, we add the audience from the config to the final url by substituting the `[AUDIENCE]`
     // with real audience string from the config.
-    std::string final_url =
-        absl::StrReplaceAll(filter_config_->http_uri().uri(), {{"[AUDIENCE]", audience_str_}});
+    std::string final_url = absl::StrReplaceAll(UrlString, {{"[AUDIENCE]", audience_str_}});
```

Comment:
This change decommissions the field, not deprecates. For example if someone had this URL configured to something other than `UrlString` value, they will get unexpected change in behavior. 

Is it possible for `http_uri` to be different from `UrlString` ? If so you would need to respect user configured value here.

Reply:
Currently filter actually doesn't support URI different from `UrlString`. So I would not expect user have different configured value here.

But to be safer, I can use runtime guard. The long-term goal is to make this URL fixed string as `UrlString`